splink/splink_dataframe.py

```diff
@@ -42,7 +43,7 @@ def physical_and_template_names_equal(self):
         return self.templated_name == self.physical_name
 
     def _check_drop_table_created_by_splink(self, force_non_splink_table=False):
-        if not self.physical_name.startswith("__splink__"):
+        if not self.physical_name.startswith(f"{self._target_schema}.__splink__"):
```

Comment:
Yes, this is related to the change in the target schema, from `public` to `splink`.

The idea is that I could isolate the effects of whatever DDL/DML splink could run in my staging backend to a new `splink` schema, since `public` contained data already.

Yet, I've forgotten to replicate the logic to the base Linker class definition - this means that all the other linkers are creating the tables in the public schema, and this code is expecting them to create the schema.

So, here is where a design decision should be made: 
- Should the package create its own schema in a backend: this would require updating schema in other linkers
- Should the package stay with public: this would solve the problem, but in this case it would be wise to be sure that the DDL/DML in the code does not interact with anything outside its scope (which I didn't check).

Reply:
It's a good question. I think probably the first option makes the most sense in a general way, but for this PR perhaps it is best to play it more conservatively and go for the third option.
I don't think the first option should cause any problems, but I feel it would be best avoid for now, so as not to make too many changes affecting the other linkers in one go, just in case.

Reply:
One more thought on this - perhaps it would be possible after creating the schema to add it to the path - something like:
```SQL
CREATE SCHEMA IF NOT EXISTS {self._db_schema};
SET search_path TO {self._db_schema},public;
```
which I _think_ should facilitate removal of all these `{self._target_schema}.` qualifiers without polluting the public schema - this is similar to how things work currently for DuckDB if a schema is provided.

There may be some good reason why this doesn't work/isn't so straightforward/is not a good idea - just a thought.